pdns/dnsmessage.proto

```diff
@@ -126,6 +126,34 @@ message PBDNSMessage {
     required MetaValue value = 2;
   }
   repeated Meta meta = 22;                      // Arbitrary meta-data - to be used in future rather than adding new fields all the time
+
+  enum EventType {
+    RecRecv = 1;
+    DistPipe = 2;
+    PCacheCheck = 3;
+    SyncRes = 4;
+    AnswerSent = 5;
+    LuaGetTag = 100;
+    LuaGetTagFFI = 101;
+    LuaIPFilter = 102;
+    LuaPreRPZ = 103;
+    LuaPreResolve = 104;
+    LuaPreOutQuery = 105;
+    LuaPostResolve = 106;
+    LuaNoData = 107;
+    LuaNXDomain = 108;
+  }
+  
+  message Event {
```

Comment:
In the future we may want to add a Lua API to add custom events. It may make sense to add a `name` field for such custom events. It would be useful to already allocate the field here, so that performance tooling can already depend on it being present.

The C++ events would not use this, because we know in advance what events are possible and it would just increase the size of the protobuf.

Reply:
Either that or reserve a numeric range for "custom events"?

Reply:
Then you would have to maintain and distribute a mapping to every component that needs it in order to interpret those custom events.  Not impossible, but the gains would need to be worth the pain.